src/Symfony/Component/Lock/Lock.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock;
+
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+use Symfony\Component\Lock\Store\BlockingStoreInterface;
+use Symfony\Component\Lock\Store\ExpirableStoreInterface;
+
+/**
+ * Lock is the default implementation to the LockInterface.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class Lock implements LockInterface
+{
+    private $store;
+    private $key;
+
+    public function __construct(Key $key, StoreInterface $store)
+    {
+        $this->store = $store;
+        $this->key = $key;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function acquire($blocking = false, $ttl = 300.0)
+    {
+        try {
+            if (!$blocking) {
+                $this->store->save($this->key);
+            } elseif ($this->store instanceof BlockingStoreInterface) {
+                $this->store->waitAndSave($this->key);
+            } else {
+                throw new InvalidArgumentException(
+                    sprintf('Unable to acquire a blocking lock on a instance of "%s".', get_class($this->store))
+                );
+            }
+            $this->refresh($ttl);
+        } catch (LockConflictedException $e) {
+            throw $e;
+        } catch (\Exception $e) {
+            throw new LockAcquiringException('', 0, $e);
+        }
```

Comment:
on Error, it should let the error throw imho. hrowing a LockAcquiringException would make no sense. Error is a dev mistake, not a runtime one.